warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:38:9
   |
LL | #![warn(x5400)] //~ WARN unknown lint: `x5400`
   |         ^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:33:28
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |                            ^^^^^^^^^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:39:10
   |
LL | #![allow(x5300)] //~ WARN unknown lint: `x5300`
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:40:11
   |
LL | #![forbid(x5200)] //~ WARN unknown lint: `x5200`
   |           ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:41:9
   |
LL | #![deny(x5100)] //~ WARN unknown lint: `x5100`
   |         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:102:8
   |
LL | #[warn(x5400)]
   |        ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:105:25
   |
LL |     mod inner { #![warn(x5400)] }
   |                         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:108:12
   |
LL |     #[warn(x5400)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:111:12
   |
LL |     #[warn(x5400)] struct S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:114:12
   |
LL |     #[warn(x5400)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:117:12
   |
LL |     #[warn(x5400)] impl S { }
   |            ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:121:9
   |
LL | #[allow(x5300)]
   |         ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:124:26
   |
LL |     mod inner { #![allow(x5300)] }
   |                          ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:127:13
   |
LL |     #[allow(x5300)] fn f() { }
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:130:13
   |
LL |     #[allow(x5300)] struct S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:133:13
   |
LL |     #[allow(x5300)] type T = S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:136:13
   |
LL |     #[allow(x5300)] impl S { }
   |             ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:140:10
   |
LL | #[forbid(x5200)]
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:143:27
   |
LL |     mod inner { #![forbid(x5200)] }
   |                           ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:146:14
   |
LL |     #[forbid(x5200)] fn f() { }
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:149:14
   |
LL |     #[forbid(x5200)] struct S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:152:14
   |
LL |     #[forbid(x5200)] type T = S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:155:14
   |
LL |     #[forbid(x5200)] impl S { }
   |              ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:159:8
   |
LL | #[deny(x5100)]
   |        ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:162:25
   |
LL |     mod inner { #![deny(x5100)] }
   |                         ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:165:12
   |
LL |     #[deny(x5100)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:168:12
   |
LL |     #[deny(x5100)] struct S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:171:12
   |
LL |     #[deny(x5100)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:174:12
   |
LL |     #[deny(x5100)] impl S { }
   |            ^^^^^

warning: macro_escape is a deprecated synonym for macro_use
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:463:1
   |
LL | #[macro_escape]
   | ^^^^^^^^^^^^^^^

warning: macro_escape is a deprecated synonym for macro_use
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:466:17
   |
LL |     mod inner { #![macro_escape] }
   |                 ^^^^^^^^^^^^^^^^
   |
   = help: consider an outer attribute, #[macro_use] mod ...

warning: `repr` attribute must have a hint
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:303:17
   |
LL |     mod inner { #![repr()] }
   |                 ^^^^^^^^^^ needs a hint
   |
   = note: #[warn(bad_repr)] on by default
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute must have a hint
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:306:5
   |
LL |     #[repr()] fn f() { }
   |     ^^^^^^^^^ needs a hint
   |
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute must have a hint
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:311:5
   |
LL |     #[repr()] type T = S;
   |     ^^^^^^^^^ needs a hint
   |
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute must have a hint
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:314:5
   |
LL |     #[repr()] impl S { }
   |     ^^^^^^^^^ needs a hint
   |
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute must have a hint
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:300:1
   |
LL | #[repr()]
   | ^^^^^^^^^ needs a hint
   |
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute must have a hint
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:51:1
   |
LL | #![repr()]
   | ^^^^^^^^^^ needs a hint
   |
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:182:5
   |
LL |     #[macro_use] fn f() { }
   |     ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:33:9
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |         ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:185:5
   |
LL |     #[macro_use] struct S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:188:5
   |
LL |     #[macro_use] type T = S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:191:5
   |
LL |     #[macro_use] impl S { }
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:198:17
   |
LL |     mod inner { #![macro_export] }
   |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:201:5
   |
LL |     #[macro_export] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:204:5
   |
LL |     #[macro_export] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:207:5
   |
LL |     #[macro_export] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:210:5
   |
LL |     #[macro_export] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:195:1
   |
LL | #[macro_export]
   | ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:217:17
   |
LL |     mod inner { #![plugin_registrar] }
   |                 ^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:222:5
   |
LL |     #[plugin_registrar] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:225:5
   |
LL |     #[plugin_registrar] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:228:5
   |
LL |     #[plugin_registrar] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:214:1
   |
LL | #[plugin_registrar]
   | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:235:17
   |
LL |     mod inner { #![main] }
   |                 ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:240:5
   |
LL |     #[main] struct S;
   |     ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:243:5
   |
LL |     #[main] type T = S;
   |     ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:246:5
   |
LL |     #[main] impl S { }
   |     ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:232:1
   |
LL | #[main]
   | ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:253:17
   |
LL |     mod inner { #![start] }
   |                 ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:258:5
   |
LL |     #[start] struct S;
   |     ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:261:5
   |
LL |     #[start] type T = S;
   |     ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:264:5
   |
LL |     #[start] impl S { }
   |     ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:250:1
   |
LL | #[start]
   | ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:322:5
   |
LL |     #[path = "3800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:325:5
   |
LL |     #[path = "3800"]  struct S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:328:5
   |
LL |     #[path = "3800"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:331:5
   |
LL |     #[path = "3800"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:338:17
   |
LL |     mod inner { #![automatically_derived] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:341:5
   |
LL |     #[automatically_derived] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:344:5
   |
LL |     #[automatically_derived] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:347:5
   |
LL |     #[automatically_derived] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:350:5
   |
LL |     #[automatically_derived] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:335:1
   |
LL | #[automatically_derived]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:370:17
   |
LL |     mod inner { #![no_link] }
   |                 ^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:373:5
   |
LL |     #[no_link] fn f() { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:376:5
   |
LL |     #[no_link] struct S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:379:5
   |
LL |     #[no_link]type T = S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:382:5
   |
LL |     #[no_link] impl S { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:367:1
   |
LL | #[no_link]
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:389:17
   |
LL |     mod inner { #![should_panic] }
   |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:392:5
   |
LL |     #[should_panic] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:395:5
   |
LL |     #[should_panic] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:398:5
   |
LL |     #[should_panic] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:401:5
   |
LL |     #[should_panic] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:386:1
   |
LL | #[should_panic]
   | ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:408:17
   |
LL |     mod inner { #![ignore] }
   |                 ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:411:5
   |
LL |     #[ignore] fn f() { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:414:5
   |
LL |     #[ignore] struct S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:417:5
   |
LL |     #[ignore] type T = S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:420:5
   |
LL |     #[ignore] impl S { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:405:1
   |
LL | #[ignore]
   | ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:427:17
   |
LL |     mod inner { #![no_implicit_prelude] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:430:5
   |
LL |     #[no_implicit_prelude] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:433:5
   |
LL |     #[no_implicit_prelude] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:436:5
   |
LL |     #[no_implicit_prelude] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:439:5
   |
LL |     #[no_implicit_prelude] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:424:1
   |
LL | #[no_implicit_prelude]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:446:17
   |
LL |     mod inner { #![reexport_test_harness_main="2900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:449:5
   |
LL |     #[reexport_test_harness_main = "2900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:452:5
   |
LL |     #[reexport_test_harness_main = "2900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:455:5
   |
LL |     #[reexport_test_harness_main = "2900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:458:5
   |
LL |     #[reexport_test_harness_main = "2900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:443:1
   |
LL | #[reexport_test_harness_main = "2900"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:469:5
   |
LL |     #[macro_escape] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:472:5
   |
LL |     #[macro_escape] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:475:5
   |
LL |     #[macro_escape] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:478:5
   |
LL |     #[macro_escape] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:486:17
   |
LL |     mod inner { #![no_std] }
   |                 ^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:486:17
   |
LL |     mod inner { #![no_std] }
   |                 ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:490:5
   |
LL |     #[no_std] fn f() { }
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:490:5
   |
LL |     #[no_std] fn f() { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:494:5
   |
LL |     #[no_std] struct S;
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:494:5
   |
LL |     #[no_std] struct S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:498:5
   |
LL |     #[no_std] type T = S;
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:498:5
   |
LL |     #[no_std] type T = S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:502:5
   |
LL |     #[no_std] impl S { }
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:502:5
   |
LL |     #[no_std] impl S { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:482:1
   |
LL | #[no_std]
   | ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:482:1
   |
LL | #[no_std]
   | ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:641:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:641:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:645:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:645:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:649:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:649:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:653:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:653:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:657:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:657:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:637:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:637:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:666:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:666:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:670:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:670:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:674:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:674:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:678:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:678:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:682:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:682:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:662:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:662:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:691:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:691:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:695:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:695:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:699:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:699:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:703:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:703:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:707:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:707:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:687:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:687:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:717:17
   |
LL |     mod inner { #![no_main] }
   |                 ^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:717:17
   |
LL |     mod inner { #![no_main] }
   |                 ^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:721:5
   |
LL |     #[no_main] fn f() { }
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:721:5
   |
LL |     #[no_main] fn f() { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:725:5
   |
LL |     #[no_main] struct S;
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:725:5
   |
LL |     #[no_main] struct S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:729:5
   |
LL |     #[no_main] type T = S;
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:729:5
   |
LL |     #[no_main] type T = S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:733:5
   |
LL |     #[no_main] impl S { }
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:733:5
   |
LL |     #[no_main] impl S { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:713:1
   |
LL | #[no_main]
   | ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:713:1
   |
LL | #[no_main]
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:755:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:755:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:759:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:759:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:763:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:763:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:767:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:767:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:771:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:771:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:751:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:751:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:780:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:780:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:784:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:784:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:788:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:788:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:792:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:792:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:796:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:796:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:776:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:776:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:43:1
   |
LL | #![macro_export] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:1
   |
LL | #![plugin_registrar] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:47:1
   |
LL | #![main] //~ WARN unused attribute
   | ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:48:1
   |
LL | #![start] //~ WARN unused attribute
   | ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:51:1
   |
LL | #![repr()]
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:54:1
   |
LL | #![path = "3800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:55:1
   |
LL | #![automatically_derived] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:57:1
   |
LL | #![no_link] //~ WARN unused attribute
   | ^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:59:1
   |
LL | #![should_panic] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:60:1
   |
LL | #![ignore] //~ WARN unused attribute
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:66:1
   |
LL | #![proc_macro_derive()] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^

error: invalid windows subsystem `1000`, only `windows` and `console` are allowed

error: aborting due to previous error

